base/reduce.jl

```diff
@@ -870,12 +870,15 @@ end
 """
     findmax(f, domain) -> (f(x), index)
 
-Returns a pair of a value in the codomain (outputs of `f`) and the index of
+Returns a pair of a value in the codomain (outputs of `f`) and the index or key of
 the corresponding value in the `domain` (inputs to `f`) such that `f(x)` is maximised.
 If there are multiple maximal points, then the first one will be returned.
 
 `domain` must be a non-empty iterable.
 
+Indices are of the same type as those returned by [`keys(domain)`](@ref)
+and [`pairs(domain)`](@ref).
```

Comment:
Should the "non-empty iterable" here also be changed? I think it's required that `keys(domain)` work, as e.g. `findmax(cbrt, x for x in -27:27 if true)` fails.

Reply:
Indeed. Something like this?
```julia
`domain` must be a non-empty iterable supporting [`pairs`](@ref).
Indices are of the same type as those returned by [`keys(domain)`](@ref)
and [`pairs(domain)`](@ref).
```

Reply:
We could leave out `pairs` altogether
```suggestion
`domain` must be a non-empty iterable supporting [`keys`](@ref). Indices
are of the same type as those returned by [`keys(domain)`](@ref).
```